sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java

```diff
@@ -119,6 +124,46 @@ public MerchantOrderStatusResponse retrieveAnnouncement(ApiNotification apiNotif
         return connector.getAnnouncementData(apiNotification);
     }
 
+    /**
+     * This function will initiate refund transaction.
+     *
+     * @param refundRequest The request for refund, the object can be constructed via Jackson or with the direct constructor
+     * @param transactionId The transactionId of transaction for which the refund request is sent
+     * @return The response contains refund details, which can be used to update the refund with the latest status.
+     * @throws RabobankSdkException when problems occurred during the request, e.g. server not reachable, invalid signature, invalid authentication etc.
+     */
+    public RefundDetailsResponse initiateRefundTransaction(InitiateRefundRequest refundRequest, UUID transactionId)
+            throws RabobankSdkException {
+        return connector.postRefundRequest(refundRequest, transactionId, tokenProvider.getAccessToken());
```

Comment:
shall we validateAccessToken() before the request ?

Reply:
Seems like an overkill to me. It's up to ROFE API to validate it, beside, we do have a logic to handle InvalidAccessTokenException and token auto-renewal (see Endpoint.class)